CoreScriptsRoot/Modules/Chat.lua

```diff
@@ -270,22 +270,30 @@ do
 		return nil -- Found no player
 	end
 
-	local adminCache = {}
-	function Util.IsPlayerAdminAsync(player)
-		local userId = player and player.userId
-		if userId then
-			if adminCache[userId] == nil then
-				local isAdmin = false
-				-- Many things can error is the IsInGroup check
-				pcall(function()
-					isAdmin = player:IsInGroup(1200769) or player:IsInGroup(2868472)
-				end)
-				adminCache[userId] = isAdmin
+	local function MakeIsInGroup(groupId, requiredRank)
+		assert(type(requiredRank) == "nil" or type(requiredRank) == "number", "requiredRank must be a number or nil")
+		
+		local inGroupCache = {}
+		return function(player)
+			if player and player.userId then
+				local userId = player.userId
+
+				if inGroupCache[userId] then
```

Comment:
Eh, did you forget a ` == nil` or am I imagining things?
It seems as if this will **always** be *false*, resulting it never being set.